File: 7. Reverse Integer.cpp
#include <string>
#include <algorithm>
class Solution {
public:
    int reverse(int x) {
        try
        {
            int sign = 1;
            if (x < 0)
            {
                x = -x;
                sign = -1;
            }
            string str_x = to_string(x);
            string reverse_str_x = "";
            for (int i = str_x.size()-1; i >= 0; i--)
                reverse_str_x += str_x[i];
            return sign == 1 ? stoi(reverse_str_x) : -stoi(reverse_str_x);
        }
        catch (...)
        {
            return 0;
        }
    }
};

File: 7. Reverse Integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        sign = 1
        if x < 0:
            x = -x
            sign = -1
        temp = int(str(x)[::-1])
        if temp >= 2147483647:
            return 0
        return temp if sign == 1 else -temp
